src/hckFetchAwsSdkHttpHandler.ts

```diff
@@ -0,0 +1,165 @@
+import { FetchHttpHandler, FetchHttpHandlerOptions, keepAliveSupport } from '@smithy/fetch-http-handler';
+import { HttpRequest, HttpResponse } from '@smithy/protocol-http';
+import { buildQueryString } from '@smithy/querystring-builder';
+import { HeaderBag, HttpHandlerOptions } from '@smithy/types';
+
+import { hckFetch } from './hckFetch';
+
+/**
+ * The aws-sdk v3+ has a modular architecture that makes it possible to configure a request handler, aka the object
+ * that's ultimately responsible for sending the HTTP requests. This module exports a factory function that creates a
+ * request handler that uses the Electron-aware hckFetch() instead of fetch().
+ */
+export function hckFetchAwsSdkHttpHandler(): FetchHttpHandler {
+  return new HckFetchAwsSdkHttpHandler();
+}
+
+/**
+ * This is a variation of the standard FetchHttpHandler. It uses hckFetch() instead of fetch().
+ * @see https://github.com/smithy-lang/smithy-typescript/blob/main/packages/fetch-http-handler/src/fetch-http-handler.ts
+ */
+class HckFetchAwsSdkHttpHandler extends FetchHttpHandler {
+  private readonly options: FetchHttpHandlerOptions;
+
+  constructor(options: FetchHttpHandlerOptions = {}) {
+    super(options);
+    this.options = options;
+  }
+
+  async handle(request: HttpRequest, { abortSignal }: HttpHandlerOptions = {}): Promise<{ response: HttpResponse }> {
+    const requestTimeoutInMs = this.options.requestTimeout;
+    const keepAlive = this.options.keepAlive === true;
+    const credentials = this.options.credentials as RequestInit['credentials'];
+
+    // if the request was already aborted, prevent doing extra work
+    if (abortSignal?.aborted) {
+      const abortError = new Error('Request aborted');
+      abortError.name = 'AbortError';
+      return Promise.reject(abortError);
```

Comment:
@thomas-jakemeyn shouldn't this just be a `throw abortError` ?

Reply:
I sticked as much as possible to the default implementation from aws-sdk. I did not try to improve it in order to ease future evolutions.